runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/control/FnApiControlClientPoolService.java

```diff
@@ -79,7 +76,7 @@ public static FnApiControlClientPoolService offeringClientsToPool(
       // discarded, which should be performed by a call to #shutdownNow. The remote caller must be
       // able to handle an unexpectedly terminated connection.
       vendedClients.add(newClient);
```

Comment:
<!--new_thread; commit:1fd121da1417624b3b84f0300648251da64b9cb5; resolved:0-->
Is this thread-safe?